clang/tools/clang-linker-wrapper/ClangLinkerWrapper.cpp

```diff
@@ -1237,6 +1191,35 @@ static Expected<StringRef> linkDevice(ArrayRef<StringRef> InputFiles,
           << "Compatible SYCL device library binary not found\n";
   }
 
+  // For NVPTX backend we need to also link libclc and CUDA libdevice.
+  if (Triple.isNVPTX()) {
+    if (Arg *A = Args.getLastArg(OPT_sycl_nvptx_device_lib_EQ)) {
+      if (A->getValues().size() == 0)
+        return createStringError(
+            inconvertibleErrorCode(),
+            "Number of device library files cannot be zero.");
+      for (StringRef Val : A->getValues()) {
+        SmallString<128> LibName(Val);
+        if (llvm::sys::fs::exists(LibName))
+          ExtractedDeviceLibFiles.emplace_back(std::string(LibName));
+        else
+          return createStringError(
+              inconvertibleErrorCode(),
+              std::string(LibName) +
+                  " SYCL device library file for NVPTX is not found.");
+      }
+    }
+  }
+
+  if (ExtractedDeviceLibFiles.empty()) {
```

Comment:
Extra check added to make sure that device library files are available when expected.

Reply:
```suggestion
  // Make sure that device library files are available.
  if (ExtractedDeviceLibFiles.empty()) {
```

Reply:
Will add this to the upcoming commit. Thanks